src/main/scala/com/normation/cfclerk/domain/HashAlgoConstraint.scala

```diff
@@ -72,11 +72,20 @@ object HashAlgoConstraint {
   /*
    * Actually do not hash the result
    */
-  object PLAIN  extends HashAlgoConstraint {
+  object PLAIN extends HashAlgoConstraint {
     override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
     override val prefix = "plain"
   }
 
+  object PreHashed extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "pre-hashed"
+  }
+
+  object SCRIPT extends HashAlgoConstraint {
+    override def hash(input:Array[Byte]) : String = new String(input, "UTF-8")
+    override val prefix = "script"
```

Comment:
Not a problem either if other hashes containing script are evalutated.

In fact that's a plain text that indicate that you want to save a script, but maybe it's not containing one.

In fact It does not indicate that the values should be evaluated as a script, but only that the user wanted to enter a script in there